File: CoWorkHub/CoWorkHub.Model/Reservation.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CoWorkHub.Model
{
    public class Reservation
    {
        public int ReservationId { get; set; }
        public int SpaceUnitId { get; set; }
        public int UsersId { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public int PeopleCount { get; set; } = 1;
        public decimal TotalPrice { get; set; }
        public string StateMachine { get; set; }
        public virtual User Users { get; set; } = null!;
        public virtual SpaceUnit SpaceUnit { get; set; } = null!;
        public bool IsDeleted { get; set; } = false;
    }
}

File: CoWorkHub/UI/coworkhub_desktop/lib/models/reservation.dart
import 'package:json_annotation/json_annotation.dart';
import 'user.dart';
import 'space_unit.dart';

part 'reservation.g.dart';

@JsonSerializable()
class Reservation {
  int reservationId;
  int spaceUnitId;
  int usersId;
  DateTime startDate;
  DateTime endDate;
  int peopleCount;
  double totalPrice;
  String stateMachine;
  bool isDeleted;
  User? users;
  SpaceUnit? spaceUnit;
  DateTime? createdAt;

  Reservation({
    required this.reservationId,
    required this.spaceUnitId,
    required this.usersId,
    required this.startDate,
    required this.endDate,
    this.peopleCount = 1,
    required this.totalPrice,
    required this.stateMachine,
    this.isDeleted = false,
    this.users,
    this.spaceUnit,
    required this.createdAt,
  });

  factory Reservation.fromJson(Map<String, dynamic> json) =>
      _$ReservationFromJson(json);
  Map<String, dynamic> toJson() => _$ReservationToJson(this);
}
